tests/api2/test_428_smb_rpc.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import urllib.parse
+import contextlib
+import pytest
+import sys
+import os
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from functions import POST, GET, DELETE, SSH_TEST, wait_on_job
+from auto_config import (
+    ip,
+    dev_test,
+    pool_name,
+    password,
+    user,
+)
+from protocols import MS_RPC
+from pytest_dependency import depends
+from utils import create_dataset
+
+reason = 'Skipping for test development testing'
+# comment pytestmark for development testing with --dev-test
+pytestmark = pytest.mark.skipif(dev_test, reason=reason)
+
+SMB_USER = "smbrpcuser"
+SMB_PWD = "smb1234#!@"
+
+
+@contextlib.contextmanager
+def smb_share(path, options=None):
+    results = POST("/sharing/smb/", {
+        "path": path,
+        **(options or {}),
+    })
+    assert results.status_code == 200, results.text
+    id = results.json()["id"]
+
+    try:
+        yield id
+    finally:
+        result = DELETE(f"/sharing/smb/id/{id}/")
+        assert result.status_code == 200, result.text
+
+    assert results.status_code == 200, results.text
+    global next_uid
+    next_uid = results.json()
+
+
+@pytest.fixture(scope="module")
+def start_smb(request):
+    results = POST("/service/start/", {"service": "cifs"})
+    assert results.status_code == 200, results.text
+
+    results = GET("/service?service=cifs")
+    assert results.json()[0]["state"] == "RUNNING", results.text
+
+    try:
+        yield request
+
+    finally:
+        results = POST("/service/stop/", {"service": "cifs"})
+        assert results.status_code == 200, results.text
+
+
+@pytest.mark.dependency(name="SMB_USER_CREATED")
```

Comment:
You can make this a fixture too, and it will allow to create and delete the user in the same method